ktor-io/common/src/io/ktor/utils/io/SourceByteReadChannel.kt

```diff
@@ -26,7 +27,7 @@ internal class SourceByteReadChannel(private val source: Source) : ByteReadChann
 
     override suspend fun awaitContent(min: Int): Boolean {
         closedCause?.let { throw it }
-        return false
+        return source.remaining >= min
```

Comment:
This isn't so much related to the change, but I noticed the logic here was wrong.